include/faabric/scheduler/MpiWorld.h

```diff
@@ -214,6 +224,8 @@ class MpiWorld
       localQueueMap;
     std::unordered_map<int, std::thread> asyncThreadMap;
 
+    std::vector<int> cartProcsPerDim;
```

Comment:
This is a bit of state that would get embedded in the communicator, but if chosen to keep the communicator simple and store it here. It stores how many MPI processes are there along each dimension (i.e. in the `i x j x 1` grid).

It is safe to do so as all processes are invoked with the same parameters.